group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -180,12 +192,19 @@ public static class DeadlineAndEpoch {
      */
     private DeadlineAndEpoch metadataRefreshDeadline = DeadlineAndEpoch.EMPTY;
 
+    /**
+     * Map of protocol names to the number of members that use legacy protocol and support them.
+     */
+    private final TimelineHashMap<String, Integer> legacyProtocolMembersSupportedProtocols;
```

Comment:
I am not sure to fully understand why we need to maintain this. Could you elaborate a bit more?

Reply:
We will need it to check if at least one of the given protocols in the `JoinGroupRequest` can be supported if a consumer using the classic protocol joins the group.